pallets/parachain-staking/src/lib.rs

```diff
@@ -2097,6 +2150,75 @@ pub mod pallet {
 			}
 		}
 
+		/// Annually reduce the reward rates for collators and delegators.
+		///
+		/// # <weight>
+		/// Weight: O(1)
+		/// - Reads: LastRewardReduction, InflationConfig
+		/// - Writes: LastRewardReduction, InflationConfig
+		/// # </weight>
+		fn adjust_reward_rates(now: T::BlockNumber) -> Weight {
+			let year = now / T::BlocksPerYear::get();
+			let last_update = <LastRewardReduction<T>>::get();
+			if year > last_update {
+				let inflation = <InflationConfig<T>>::get();
+				// collator reward rate decreases by 2% of the previous one per year
+				let c_reward_rate = inflation.collator.reward_rate.annual * Perquintill::from_percent(98);
+				// delegator reward rate should be 6% in 2nd year and 0% afterwards
+				let d_reward_rate = if year == T::BlockNumber::one() {
+					Perquintill::from_percent(6)
+				} else {
+					Perquintill::zero()
+				};
+
+				let new_inflation = InflationInfo::new::<T>(
+					inflation.collator.max_rate,
+					c_reward_rate,
+					inflation.delegator.max_rate,
+					d_reward_rate,
+				);
+				<InflationConfig<T>>::put(new_inflation.clone());
+				<LastRewardReduction<T>>::put(year);
+				Self::deposit_event(Event::RoundInflationSet(
+					new_inflation.collator.max_rate,
+					new_inflation.collator.reward_rate.per_block,
+					new_inflation.delegator.max_rate,
+					new_inflation.delegator.reward_rate.per_block,
+				));
+				<T as Config>::WeightInfo::on_initialize_new_year();
+			}
+			T::DbWeight::get().reads(1)
+		}
+
+		/// Checks whether a delegator can still delegate in this round, e.g.,
+		/// if they have not delegated MaxDelegationsPerRound many times
+		/// already in this round.
+		///
+		/// # <weight>
+		/// Weight: O(1)
+		/// - Reads: LastDelegation, Round
+		/// # </weight>
+		fn update_delegation_counter(delegator: &T::AccountId) -> Result<DelegationCounter, DispatchError> {
```

Comment:
The name `update` is not really correct here as all this function does is to check whether the delegator can still delegate.

```suggestion
		fn get_delegation_counter(delegator: &T::AccountId) -> Result<DelegationCounter, DispatchError> {
```